python/cugraph/cugraph/tests/link_prediction/test_jaccard.py

```diff
@@ -144,187 +173,181 @@ def read_csv(request):
+    if use_weight:
+        cu_M[EDGE_ATT_COL] = cudf.Series(M[EDGE_ATT_COL])
+
     G1 = cugraph.Graph()
+    weight = EDGE_ATT_COL if use_weight else None
     G1.from_cudf_edgelist(
-        cu_M, source=["src_0", "src_1"], destination=["dst_0", "dst_1"]
+        cu_M,
+        source=[MULTI_COL_SRC_0_COL, MULTI_COL_SRC_1_COL],
+        destination=[MULTI_COL_DST_0_COL, MULTI_COL_DST_1_COL],
+        weight=weight,
     )
 
-    vertex_pair = cu_M[["src_0", "src_1", "dst_0", "dst_1"]]
+    vertex_pair = cu_M[
+        [
+            MULTI_COL_SRC_0_COL,
+            MULTI_COL_SRC_1_COL,
+            MULTI_COL_DST_0_COL,
+            MULTI_COL_DST_1_COL,
+        ]
+    ]
     vertex_pair = vertex_pair[:5]
 
-    df_res = cugraph.jaccard(G1, vertex_pair)
-    df_plc_exp = exp_jaccard(G1, vertex_pair)
-
-    df_plc_exp = df_plc_exp.rename(
-        columns={
-            "0_src": "0_source",
-            "0_dst": "0_destination",
-            "1_src": "1_source",
-            "1_dst": "1_destination",
-        }
-    )
-
-    jaccard_res = df_res["jaccard_coeff"].sort_values().reset_index(drop=True)
-    jaccard_plc_exp = df_plc_exp["jaccard_coeff"].sort_values().reset_index(drop=True)
-    assert_series_equal(jaccard_res, jaccard_plc_exp)
+    df_multi_col_res = cugraph.jaccard(G1, vertex_pair)
 
     G2 = cugraph.Graph()
-    G2.from_cudf_edgelist(cu_M, source="src_0", destination="dst_0")
-    df_exp = cugraph.jaccard(G2, vertex_pair[["src_0", "dst_0"]])
+    G2.from_cudf_edgelist(
+        cu_M, source=MULTI_COL_SRC_0_COL, destination=MULTI_COL_DST_0_COL, weight=weight
+    )
+    df_single_col_res = cugraph.jaccard(
+        G2, vertex_pair[[MULTI_COL_SRC_0_COL, MULTI_COL_DST_0_COL]]
+    )
 
     # Calculating mismatch
-    actual = df_res.sort_values("0_first").reset_index()
-    expected = df_exp.sort_values("first").reset_index()
-    assert_series_equal(actual["jaccard_coeff"], expected["jaccard_coeff"])
+    actual = df_multi_col_res.sort_values("0_src").reset_index()
+    expected = df_single_col_res.sort_values(VERTEX_PAIR_FIRST_COL).reset_index()
+    assert_series_equal(actual[JACCARD_COEFF_COL], expected[JACCARD_COEFF_COL])
 
 
 @pytest.mark.sg
-def test_weighted_exp_jaccard():
+def test_weighted_jaccard():
     karate = UNDIRECTED_DATASETS[0]
-    G = karate.get_graph()
-    with pytest.raises(ValueError):
-        exp_jaccard(G)
-
     G = karate.get_graph(ignore_weights=True)
-    use_weight = True
     with pytest.raises(ValueError):
-        exp_jaccard(G, use_weight=use_weight)
-
+        cugraph.jaccard(G, use_weight=True)
 
-@pytest.mark.sg
-def test_invalid_datasets_jaccard():
-    karate = UNDIRECTED_DATASETS[0]
-    df = karate.get_edgelist()
-    df = df.add(1)
-    G = cugraph.Graph(directed=False)
-    G.from_cudf_edgelist(df, source="src", destination="dst")
+    G = karate.get_graph(ignore_weights=True)
     with pytest.raises(ValueError):
-        cugraph.jaccard(G)
+        cugraph.jaccard(G, use_weight=True)
```

Comment:
Isn't this just testing the same condition twice?

Reply:
Yes this is repetitive. If you look at the old tests on the left. you will see there were two checks for `ValueError`. He probably forgot to take one of them out as it is outdated. For the second check, I believe the C or C++ API will raise a `ValueError` if the user didn't pass weights at the graph creation but set the argument `use_weight=True`. I am wondering if we should check for that as well at the python layer. Something like:
```
if G.is_weighted() and use_weight:
    raise ValueError(
        "'use_weight' is set to 'True' but there are no weights in the graph")
```